src/pgwire/src/server.rs

```diff
@@ -138,7 +140,18 @@ impl Server {
                                 conn_uuid_handle.set(conn_uuid);
                                 debug!(conn_uuid = %conn_uuid_handle.display(), conn_uuid_forwarded, "starting new pgwire connection in adapter");
 
-                                let mut conn = FramedConn::new(conn_id.clone(), conn);
+                                let direct_peer_addr = conn.inner_mut().peer_addr().map(|addr| addr.ip()).expect("pgwire connection must have peer address");
+                                let peer_addr= params
+                                    .remove(MZ_FORWARD_FOR_KEY)
+                                    .and_then(|ip| IpAddr::from_str(&ip).inspect_err(|e| error!("pgwire connection with invalid peer address: {e}"))
```

Comment:
I feel like if the forwarded IP exists but is an invalid address, we should use `None` instead of falling back to the direct peer address, since we were obviously trying to connect via the balancer and just messed it up somehow, and the direct peer address is going to be the balancer IP which is probably not what we want to expose.